src/Typography/Typography.component.tsx

```diff
@@ -139,13 +139,13 @@ const StyledTypography = (tag: TypographyTags) => styled(tag)<
 export const Typography = ({
     className,
     children,
-    tag = 'span',
+    tag: as = 'span',
     color,
     ...props
 }: TypographyProps): JSX.Element =>
-    React.createElement(StyledTypography(tag), {
+    React.createElement(StyledTypography(as), {
```

Comment:
I was aiming for a partial refactor here, but it may make more sense for this to be a breaking change. Styled-Comps has supported a global `as` prop since V4 and we just don't need this anymore.

Reply:
https://styled-components.com/docs/api#as-polymorphic-prop